blaze/mahimahi/server/injected-javascript/3-find-in-viewport.js

```diff
@@ -0,0 +1,126 @@
+}
+
+
+function findAndPrintImagesInViewport(ele) {
+    console.log("inside find and print images")
+    ele.querySelectorAll('*').forEach(function(node) {
+        try {
+            if (isElementInViewport(node)) {
+                var url = null;
+                if(node.tagName == "IMG") {
+                    if (typeof node.href != 'undefined') {
+                        url = node.href;
+                    }
+                    if(typeof node.src != 'undefined') {
+                        url = node.src;
+                    }
+                    if (url != null) {
+                        // console.log("image is in viewport ", url)
+                        imagesInViewPort.push(url)
+                    }
+            } else {
+                var style = window.getComputedStyle(node)
+                for (var i = style.length - 1; i >= 0; i--) {
+                    var cssName = style[i]
+                    var cssPropertyValue = style.getPropertyValue(cssName)
+                    if (cssPropertyValue.indexOf("url") >= 0) {
+                        var potentialURL = cssPropertyValue;
+                        var startIndex = potentialURL.indexOf('url(')
+                        var urlWithSpace = false;
+                        if (startIndex < 0) {
+                            startIndex = potentialURL.indexOf('url (')
+                            urlWithSpace = true;
+                        } 
+                        if (startIndex < 0) {
+                            continue;
+                        }
+                        var endIndex = potentialURL.indexOf(')', startIndex)
+                        if (endIndex < 0) {
+                            continue
+                        }
+                        var potentialURL = potentialURL.substring(startIndex + (urlWithSpace ? 5 : 4), endIndex)
+                        var t=potentialURL.length;
+                        if (potentialURL.charAt(0)=='"'||potentialURL.charAt(0)=="'") {
+                            potentialURL = potentialURL.substring(1);
+                            t--;
+                        }
+                        if (potentialURL.charAt(t-1)=='"'||potentialURL.charAt(t-1)=="'") {
+                            potentialURL = potentialURL.substring(0,t-1);
+                        }
+    
+                        var link = document.createElement("a");
+                        link.href = potentialURL;
+                        if(potentialURL.index("data:image") < 0)
+                            imagesInViewPort.push(link.href)
+                    }
+                }
+            }  
+            }           
+        } catch (error) {
+            console.log("ignoring node due to error ", error)
+        }
+
+    });
+    var answer = getCriticalRequests()
+    console.log("alohomora: this is the answer")
+    console.log(JSON.stringify({'alohomora_output': answer}))
+    console.log("alohomora: that was the the answer")
```

Comment:
remove

Reply:
removed "alohomora: that was the the answer".
we need the other one.